src/frontend/src/reducers/neighborhoods/neighborhoodsSlice.ts

```diff
@@ -118,6 +119,38 @@ export const selectAreFiltersApplied = (state: RootState) => {
 export const selectNeighborhoodFilters = (state: RootState) =>
     state.neighborhoods.filters;
 
+export const selectNeighborhoodFilterSuggestions = createSelector(
+    [
+        selectRankedNeighborhoodsLists,
+        (state: RootState) => state.neighborhoods.neighborhoods,
+    ],
+    (rankedNeighborhoodsLists, neighborhoods) => {
+        const { topTen, recommended, tooFar } = rankedNeighborhoodsLists;
+        // Return distinct town names formatted as Autocomplete Suggestion
+        return neighborhoods?.features
+            .reduce(
+                (
+                    towns: string[],
+                    f: Feature<Point, NeighborhoodProperties>
+                ) => {
+                    const isDistinct = !towns.includes(f.properties.town);
+                    const isInFilteredList = [
```

Comment:
I'd suggest creating a `filteredList` before the reduce, since this list won't change in each iteration, so we could avoid spreading and creating a new list in each iteration.